test/functorch/test_control_flow.py

```diff
@@ -45,8 +54,101 @@ def f(x, y):
         xs = torch.ones(3, 2, 2, device="cuda")
         y = torch.ones(2, device="cuda")
         res = control_flow.map(f, xs, y)
+        expected = _fake_map(f, xs, y)
+        self.assertEqual(expected, res)
+
+    def test_map_illegal_inputs(self):
+        def f(x, y):
+            return x[0] + x[1] + y
+
+        with self.assertRaisesRegex(RuntimeError,
+                                    r"Mapped xs can only consist of tensors\. Got xs \[3, tensor\(\[1\., 1\.\]\)\]\."):
+            _ = control_flow.map(f, (3, torch.ones(2)), torch.ones(2))
+
+        with self.assertRaisesRegex(RuntimeError,
+                                    r"Leading dimensions of mapped xs cannot be 0\."):
+            _ = control_flow.map(f, (torch.ones(0, 1, 2), torch.ones(0, 1, 2)), torch.ones(2))
+
+        with self.assertRaisesRegex(RuntimeError,
+                                    r"Leading dimensions of mapped xs must be consistent\. "
+                                    r"Got shapes \[torch\.Size\(\[3, 4, 5\]\), torch\.Size\(\[4, 4, 5\]\)\]\."):
+            _ = control_flow.map(f, (torch.ones(3, 4, 5), torch.ones(4, 4, 5)), torch.ones(5))
+
```

Comment:
Can we also test the output of `f` needs to be a Tensor?

Reply:
Sure! Test added.